go/services/reports/pkg/incident_correlation/utils.go

```diff
@@ -0,0 +1,100 @@
+)
+
+func ConvertConcreteIncidentArrayIntoIncidents[T repositories.Incident](incidents []T) []repositories.Incident {
+	convertedIncidents := make([]repositories.Incident, 0, len(incidents))
+
+	for _, incident := range incidents {
+		convertedIncidents = append(convertedIncidents, incident)
+	}
+
+	return convertedIncidents
+}
+
+// Removes merged incidents from report and returns new incident as a result
+func mergeApplicationIncidents(group IncidentMergeGroup, applicationIncidents *repositories.ApplicationReport) *repositories.ApplicationIncident {
+	var sourcesToBeMerged []repositories.ApplicationIncidentSource
+	var filteredIncidents []*repositories.ApplicationIncident
+
+	for _, incident := range applicationIncidents.Incidents {
+		if !slices.Contains(group.IncidentIds, incident.Id) {
+			filteredIncidents = append(filteredIncidents, incident)
+			continue
+		}
+
+		sourcesToBeMerged = append(sourcesToBeMerged, incident.Sources...)
+	}
+
+	newIncident := &repositories.ApplicationIncident{
+		Title:           group.Title,
+		Urgency:         group.Urgency,
+		Summary:         group.Summary,
+		Recommendation:  group.Recommendation,
+		Category:        group.Category,
+		CustomPrompt:    applicationIncidents.CustomPrompt,
+		Accuracy:        applicationIncidents.Accuracy,
+		ApplicationName: applicationIncidents.ApplicationName,
+		ClusterId:       applicationIncidents.Incidents[0].ClusterId,
+		Sources:         sourcesToBeMerged,
+	}
+
+	applicationIncidents.Incidents = filteredIncidents
+
+	return newIncident
+}
+func MergeApplicationIncidentsByGroups(groups []IncidentMergeGroup, applicationIncidents *repositories.ApplicationReport) []*repositories.ApplicationIncident {
+	var newIncidents []*repositories.ApplicationIncident
+	for _, group := range groups {
+		newIncident := mergeApplicationIncidents(group, applicationIncidents)
+		newIncidents = append(newIncidents, newIncident)
+	}
+
+	return newIncidents
+}
+
+// Removes merged incidents from report and returns new incident as a result
+func mergeNodeIncidents(group IncidentMergeGroup, nodeIncidents *repositories.NodeReport) *repositories.NodeIncident {
+	var sourcesToBeMerged []repositories.NodeIncidentSource
+	var filteredIncidents []*repositories.NodeIncident
+
+	for _, incident := range nodeIncidents.Incidents {
+		if !slices.Contains(group.IncidentIds, incident.Id) {
+			filteredIncidents = append(filteredIncidents, incident)
+			continue
+		}
+
+		sourcesToBeMerged = append(sourcesToBeMerged, incident.Sources...)
+	}
+
+	newIncident := &repositories.NodeIncident{
+		Title:          group.Title,
+		Urgency:        group.Urgency,
+		Summary:        group.Summary,
+		Recommendation: group.Recommendation,
+		Category:       group.Category,
+		CustomPrompt:   nodeIncidents.CustomPrompt,
+		Accuracy:       nodeIncidents.Accuracy,
+		NodeName:       nodeIncidents.Node,
+		ClusterId:      nodeIncidents.Incidents[0].ClusterId,
```

Comment:
Same here

Reply:
This is not possible, as an incident group will always have at least one element